lib/rules/no-return-value-from-each-iteratee.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileoverview Rule to check the iteratee in a call to _.each does not return a value
+ */
+'use strict';
+
+var _ = require('lodash');
+var lodashUtil = require('../util/lodashUtil');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    return {
+        ReturnStatement: 
+        function (node) {
+            function getSurroundingFunction(astNode) {
+                var ancestors = context.getAncestors();
+                var current = astNode;
+                while (!_.isUndefined(current)) {
+                    if (current.type === 'FunctionExpression' ||
+                        current.type === 'FunctionDeclaration') {
+                        break;
+                    }
+                    current = ancestors.pop();
+                }
+                return current;
+            }
+
+            function isReturnFromIteratee(astNode) {
+                var surroundingFunction = getSurroundingFunction(astNode);
+                return !_.isUndefined(surroundingFunction) &&
+                    lodashUtil.isCallToMethod(surroundingFunction.parent, 'each') &&
+                    lodashUtil.getLodashIteratee(surroundingFunction.parent) === surroundingFunction;
+            }
```

Comment:
You could move these helper function declarations outside of the `ReturnStatement` callback. That way you're not having to define them on every `return`. Here's an example of a rule that has some helper functions: https://github.com/captbaritone/eslint-plugin-underscore/blob/master/lib/rules/prefer-matches.js